src/mpl/src/gpu/mpl_gpu_ze.c

```diff
@@ -52,6 +52,42 @@ static uint32_t local_ze_device_count;  /* Counts all local devices and subdevic
 static uint32_t global_ze_device_count; /* Counts all global devices and subdevices */
 static int max_dev_id;          /* Does not include subdevices */
 static int max_subdev_id;
+static int global_dev_count;
+static int global_subdev_count;
+/* global_to_local_map[] are ordered (dev_id, sub_id), where sub_id is 0 for root device and
+ * (subdev_id + 1) for tiles */
+#define SET_GLOBAL_ZE_DEVICE_COUNT \
+    do { \
+        global_dev_count = max_dev_id + 1; \
+        global_subdev_count = max_subdev_id + 1; \
+        global_ze_device_count = global_dev_count * (global_subdev_count); \
+    } while (0)
```

Comment:
Since this should only be called once, would recommend moving the code directly to calling function instead of defined as a macro

Reply:
The purpose of macro is to move all logic of constructing global device id and global device index together for ease of maintenance.